app/cast_utils.py

```diff
@@ -124,17 +124,24 @@ def create_hibernation_node(cluster_id: str, castai_api_token: str, instance_typ
         return nodeId
 
 
-def delete_all_pausable_nodes(cluster_id: str, castai_api_token: str, hibernation_node_id: str, job_node_id=None):
+def delete_all_pausable_nodes(cluster_id: str, castai_api_token: str, hibernation_node_id: str, protect_eviction_disabled: bool, job_node_id=None):
     """" Delete all nodes through CAST AI mothership excluding hibernation node"""
     node_list_result = get_castai_nodes(cluster_id, castai_api_token)
-
+    logging.info(protect_eviction_disabled)
     for node in node_list_result["items"]:
+        skip=False
+        if "autoscaling.cast.ai/removal-disabled" in node["labels"] and protect_eviction_disabled=="true":
```

Comment:
if "autoscaling.cast.ai/removal-disabled" in node["labels"] and protect_eviction_disabled:

protect_eviction_disabled is bool meaning == True, not string "true".

Reply:
Updated